### PR TITLE
Fix missing '[' from conditioinal in bootstrap.sh

### PR DESCRIPTION
**Pull request by juwalter**
- #### What does your PR fix?  
  Fixes missing '[' from conditional in bootstrap.sh

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `have not - very minor change only `

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 no, not updated

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**
no

### CODE REVIEW

File: scripts/bootstrap.sh [hunk 1]
```diff
@@ -165,7 +165,7 @@ fetchTool()
     xmlFileAsString=`cat "$vcpkgRootDir/scripts/vcpkgTools.xml"`
     toolRegexStart="<tool name=\"$tool\" os=\"$os\">"
     toolData="$(extractStringBetweenDelimiters "$xmlFileAsString" "$toolRegexStart" "</tool>")"
-    if [ "$toolData" = "" ] || [[ "$toolData" == "<?xml"* ]]; then
+    if [[ "$toolData" = "" ] || [[ "$toolData" == "<?xml"* ]]; then
```
Comment (Neumann-A) [inline scripts/bootstrap.sh:168]:
```suggestion
    if [[ "$toolData" = "" ]] || [[ "$toolData" == "<?xml"* ]]; then
```
for me this looks like also missing a second ] ?

Comment (juwalter) [reply scripts/bootstrap.sh:168]: good catch - thanks!

**Comment by BillyONeal** [comment]
This isn't the correct fix, see 
https://github.com/microsoft/vcpkg/discussions/19822
and
https://github.com/microsoft/vcpkg/pull/19911

**Comment by juwalter** [comment]
thanks @BillyONeal - wasn't aware of the issue & discussion 

#19911 seems a much better fix

btw - running https://github.com/microsoft/vcpkg/blob/master/scripts/bootstrap.sh through https://github.com/koalaman/shellcheck reveals another `[[  ]]`

```

In bootstrap.sh line 272:
    if [ "$vcpkgAllowAppleClang" = "true" ] || [[ $(sw_vers -productVersion | awk -F '.' '{print $1}') -ge 11 ]]; then
                                               ^-- SC2039: In POSIX sh, [[ ]] is undefined.

```

https://github.com/microsoft/vcpkg/blob/master/scripts/bootstrap.sh#L272

Perhaps _shellcheck_ (or similar) can be included  in CI? https://github.com/koalaman/shellcheck#in-your-build-or-test-suites

however, there is a caveat in _shellcheck_'s README 

> Github (only Linux)

https://github.com/koalaman/shellcheck#in-your-build-or-test-suites

**Comment by BillyONeal** [comment]
I think by the time we would deploy shellcheck we will have binary releases working which will eliminate effectively everything in bootstrap.sh.

**Pull request closed without merging**